### PR TITLE
fix: dont send logs under id 0 to the api

### PR DESCRIPTION
**Pull request by lpereiracgn**

### CODE REVIEW

**Review by abdullah-cognite — APPROVED** [review]

File: Cognite.Simulator.Utils/ScopedRemoteApiSink.cs [hunk 1]
```diff
@@ -96,11 +96,13 @@ private async Task SendToRemoteApi(SimulatorsResource client, CancellationToken
...
-                            logData,
-                            token
-                        ).ConfigureAwait(false);
+                        if(log.Key != 0){
+                            await client.UpdateLogsBatch(
+                                log.Key,
+                                logData,
+                                token
+                            ).ConfigureAwait(false);   
+                        }
```
Comment (polomani) [inline Cognite.Simulator.Utils/ScopedRemoteApiSink.cs:105]: can we please not assign 0 and make a value nullable or smth?
defaulting things to 0 make these sort of bugs to happen in the first place

**Review by polomani — CHANGES_REQUESTED** [review]
make the param optional in case it is optional

File: Cognite.Simulator.Utils/ScopedRemoteApiSink.cs (hunk 1 above)
Comment (polomani) [reply Cognite.Simulator.Utils/ScopedRemoteApiSink.cs:105]: the defaulting logic happens somewhere in the library as well I assume

**Review by abdullah-cognite — APPROVED** [review]

**Comment by abdullah-cognite** [comment]
@lpereiracgn test needs a tiny fix

**Comment by polomani** [comment]
@lpereiracgn can you explain in what case we have zeros in this part of the code?

**Comment by abdullah-cognite** [comment]
> @lpereiracgn can you explain in what case we have zeros in this part of the code?

in case logId was mistakenly set to an empty string by any other part of the code.

**Review by polomani — APPROVED** [review]
LGTM

**Pull request merged**